### PR TITLE
Reinstate pool tests

### PR DESCRIPTION
**Pull request by Ben-PH**
* [x] document all added functions
* [ ] try in sandbox /simulation/labnet
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [x] add logs allowing easy debugging in case the changes caused problems
* [x] if the API has changed, update the API specification

### CODE REVIEW

**Comment by Ben-PH** [comment]
67d2921760f420f0e22c5d82032aeb15ba5a5dcc adds some `dbg!`s, and this is the typical output:

```
     Running unittests src/lib.rs (target/debug/deps/massa_pool_worker-c92c862378c0b4ac)

running 1 test
[massa-pool-worker/src/tests/operation_pool_tests.rs:129] thread = 0
[massa-pool-worker/src/operation_pool.rs:224] &slot = Slot {
    period: 31,
    thread: 0,
}
[massa-execution-exports/src/test_exports/mock.rs:217] "never seen this not err..." = "never seen this not err..."
[massa-execution-exports/src/test_exports/mock.rs:218] response_rx.recv_timeout(Duration::from_millis(100)) = Err(
    Timeout,
)
thread 'tests::operation_pool_tests::test_pool' panicked at 'called `Result::unwrap()` on an `Err` value: Timeout', massa-execution-exports/src/test_exports/mock.rs:218:68
```

Still need to work out the source of the problem...

**Comment by Ben-PH** [comment]
At this point, two of the methods are unignored, but the third one is not clear for me yet. @AurelienFT what do you think of merging these two so far, and do the rest in a follow up PR. It at least makes these two available in the meantime.

**Comment by Ben-PH** [comment]
Basically, needed to comment out the balance check to get things through.

**Comment by Ben-PH** [comment]
test is flakey: run this inside `massa-pool-worker` to (eventually) get a fail;

`while cargo test test_pool; do :; done`

**Comment by AurelienFT** [comment]
I didn't follow well everything that has been going on but to give you maybe some context that might help you. We always run the test at root with feature testing (to have exports only when test cfg (no other way with cargo yet)).

**Comment by Ben-PH** [comment]
updated script:

```sh
COUNTER = 0;
while cargo test test_pool -F testing --release ; do
    COUNTER=$[$COUNTER +1]
    :
done

echo $COUNTER;
```

@Eitu33 had a look at it with me, and it seems that this comes down to concurrent nature of the way the tests are setup?

The means to fix it is similar to the other two tests: Use the mock-receiver to intercept requests for data, and reply with hardcoded/intercepted data.

With that in mind, I'll do the final tidy up then mark for review.

File: massa-pool-worker/src/tests/scenario.rs [hunk 1]
```diff
@@ -114,12 +123,22 @@ fn launch_basic_get_block_operation_execution_mock(
         use ControllerMsg::GetFinalAndCandidateBalance as GetFinal;
         use ControllerMsg::UnexecutedOpsAmong as Unexecuted;
```
Comment (Eitu33) [inline massa-pool-worker/src/tests/scenario.rs:124]: Not related to these changes but this is a confusing and useless rename

**Review by Eitu33 — APPROVED** [review]

File: massa-pool-worker/src/tests/scenario.rs (hunk 1 above)
Comment (Ben-PH) [reply massa-pool-worker/src/tests/scenario.rs:124]: I've added a point to the follow up issue #3506 . That sufficient?

**Pull request merged**